packages/react-aria-components/test/Disclosure.test.js

```diff
@@ -206,6 +210,92 @@ describe('Disclosure', () => {
     expect(disclosure).not.toHaveAttribute('data-focus-visible-within');
     expect(disclosure).not.toHaveClass('focus');
   });
+
+  it('should support interactive elements adjacent to heading', async () => {
+    const {getByTestId, queryByText} = render(
+      <Disclosure data-testid="disclosure">
+        <Heading level={3}>
+          <Button slot="trigger" data-testid="disclosure-trigger">Trigger</Button>
+        </Heading>
+        <MenuTrigger>
+          <Button aria-label="Menu" data-testid="menu-trigger">☰</Button>
+          <Popover>
+            <Menu data-testid="menu">
+              <MenuItem id="open">Open</MenuItem>
+              <MenuItem id="rename">Rename…</MenuItem>
+              <MenuItem id="duplicate">Duplicate</MenuItem>
+              <MenuItem id="share">Share…</MenuItem>
+              <MenuItem id="delete">Delete…</MenuItem>
+            </Menu>
+          </Popover>
+        </MenuTrigger>
+        <DisclosurePanel>
+          <p>Content</p>
+        </DisclosurePanel>
+      </Disclosure>
+    );
+    const disclosure = getByTestId('disclosure');
+    const panel = queryByText('Content');
+    const menuTrigger = getByTestId('menu-trigger');
+    expect(disclosure).not.toHaveAttribute('data-expanded');
+    expect(panel).not.toBeVisible();
+    
+    await user.click(menuTrigger);
+    const menu = getByTestId('menu');
+    expect(menu).toBeVisible();
+    expect(disclosure).not.toHaveAttribute('data-expanded');
+    expect(panel).not.toBeVisible();
+    await user.click(document.body);
+    expect(menu).not.toBeVisible();
+
+    const button = getByTestId('disclosure-trigger');
+    await user.click(button);
+
+    expect(disclosure).toHaveAttribute('data-expanded', 'true');
+    expect(panel).toBeVisible();
+    expect(menu).not.toBeVisible();
+
+    await user.click(button);
+
+    expect(disclosure).not.toHaveAttribute('data-expanded');
+    expect(panel).not.toBeVisible();
+    expect(menu).not.toBeVisible();
+  });
+
+  it('should support nested Disclosures', async () => {
```

Comment:
Do RSP or S2 do anything where we would want to test nesting there too? Should this be a story in storybook or/and chromatic?

Reply:
I would say not; I think Spectrum would discourage it. Mostly wanted to have this story in RAC to make sure the contexts work properly in this case. Happy to add them if we think there are legitimate use cases though.